parachains/pallets/bridge-transfer/src/lib.rs

```diff
@@ -0,0 +1,1596 @@
+		pub fn add_reserve_location(
+			origin: OriginFor<T>,
+			location: Box<VersionedMultiLocation>,
+		) -> DispatchResult {
+			let _ = T::AdminOrigin::ensure_origin(origin)?;
+
+			let location: MultiLocation =
+				(*location).try_into().map_err(|_| Error::<T>::UnsupportedDestination)?;
+			let added = AllowedReserveLocations::<T>::try_mutate(|locations| {
+				locations.try_insert(location)
+			})
+			.map_err(|_| Error::<T>::InvalidConfiguration)?;
+			if added {
+				Self::deposit_event(Event::ReserveLocationAdded);
+			}
+			Ok(())
+		}
+
+		/// Remove `MultiLocation` mapping from `AllowedReserveLocations`
+		///
+		/// Parameters:
+		///
+		/// * `location`: as reserve `MultiLocation`
+		#[pallet::call_index(8)]
+		#[pallet::weight(T::WeightInfo::remove_reserve_location())]
+		pub fn remove_reserve_location(
+			origin: OriginFor<T>,
+			locations_to_remove: sp_std::prelude::Vec<VersionedMultiLocation>,
+		) -> DispatchResult {
+			let _ = T::AdminOrigin::ensure_origin(origin)?;
+
+			let removed =
+				AllowedReserveLocations::<T>::try_mutate(|locations| -> Result<bool, Error<T>> {
+					let mut removed = false;
+					for ltr in locations_to_remove {
+						let ltr: MultiLocation =
+							ltr.try_into().map_err(|_| Error::<T>::UnsupportedDestination)?;
+						removed |= locations.remove(&ltr);
+					}
+					Ok(removed)
+				})?;
+			if removed {
+				Self::deposit_event(Event::ReserveLocationRemoved);
+			}
+			Ok(())
+		}
+	}
+
+	impl<T: Config> Pallet<T> {
+		/// Validates destination and check if we support bridging to this remote global consensus
+		///
+		/// Returns: correct remote location, where we should be able to bridge
+		pub(crate) fn ensure_remote_destination(
+			remote_destination: VersionedMultiLocation,
+		) -> Result<(NetworkId, BridgeConfig, MultiLocation), Error<T>> {
+			match remote_destination {
+				VersionedMultiLocation::V3(remote_location) => {
+					ensure!(
+						remote_location.parent_count() == 2,
+						Error::<T>::UnsupportedDestination
+					);
+					let local_network = T::UniversalLocation::get()
+						.global_consensus()
+						.map_err(|_| Error::<T>::InvalidConfiguration)?;
+					let remote_network = remote_location
+						.interior()
+						.global_consensus()
+						.map_err(|_| Error::<T>::UnsupportedDestination)?;
+					ensure!(local_network != remote_network, Error::<T>::UnsupportedDestination);
+					match AllowedExporters::<T>::get(remote_network) {
+						Some(bridge_config) => {
+							ensure!(
+								// TODO:check-parameter - verify and prepare test for ETH scenario - https://github.com/paritytech/cumulus/pull/2013#discussion_r1094909290
+								remote_location.starts_with(&bridge_config.allowed_target_location),
+								Error::<T>::UnsupportedDestination
+							);
+							Ok((remote_network, bridge_config, remote_location))
+						},
+						None => return Err(Error::<T>::UnsupportedDestination),
+					}
+				},
+				_ => Err(Error::<T>::UnsupportedDestination),
```

Comment:
Nit: I think this could be confusing. Something like `UnuspportedXcmVersion` would be more accurate.

Reply:
@serban300 
good point, please, check also other version conversion at least I see:
```
ltr.try_into().map_err(|_| Error::<T>::UnsupportedDestination)?;
(*location).try_into().map_err(|_| Error::<T>::UnsupportedDestination)?;
```

and maybe there is one similar:
```
(*assets).try_into().map_err(|()| Error::<T>::InvalidAssets)?;
```

and also maybe we could be more concrete instead of `Error::<T>::UnsupportedDestination` (where it make sense)

Reply:
Done. Changed to `UnsupportedXcmVersion` where possible. Not Sure if I covered everything.